### ISSUE TITLE
dropdown: screen reader not announcing role of a dropdown.

### ISSUE THREAD

**Issue by arunarokk** [OfficeDev/office-ui-fabric-react#5752]
### Bug Report 
 
- __Package version(s)__: (6.38.0)
- __Browser and OS versions__: (Edge & win 10)
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (No)
 
Requested priority: (High) 
 
Products/sites affected: (if applicable)
 
#### Describe the issue:
 
screen reader not announcing role of a dropdown. it should announce as listbox.
 
https://developer.microsoft.com/en-us/fabric#/components/dropdown

**Comment by cliffkoh** [comment]
Hey @arunarokk, just to confirm, you did not encounter this with the website but with 6.38.0 version of Fabric?

**Comment by arunarokk** [comment]
@cliffkoh , i'm able to repro this here too 
https://developer.microsoft.com/en-us/fabric#/components/dropdown

**Comment by cliffkoh** [comment]
The website is extremely outdated unfortunately and does not have a whole bunch of fixes that went in recently.

In 6.38.0, the role="listbox" should be on the Dropdown. It would be great if you could be more specific about which Dropdown element (since there are multiple on the example page), what it is currently reading and what you're expecting.

**Comment by cliffkoh** [comment]
Please test it with the Dropdown here: http://fabricweb.z5.web.core.windows.net/pr/refs/pull/5756/merge/#/examples/dropdown

**Comment by arunarokk** [comment]
I have tested drop down below
http://fabricweb.z5.web.core.windows.net/pr/refs/pull/5756/merge/#/examples/dropdown

![capture](https://user-images.githubusercontent.com/38989707/43547754-0b9acaf6-9591-11e8-9aef-65305ce7fa46.JPG)

Screen reader announces Option a 1 of 10 selected.  it should also announce its role and state as well.
the expected result is  Option a 1 of 10 selected editable combo box collapsed.

**Comment by cliffkoh** [comment]
Hi @arunarokk, the dropdown in question does have the `role=listbox` and `aria-expanded=false` attributes on it. Unless there is a specific fix that Fabric can make here to be more spec compliant, please instead open an issue with Narrator or Edge per https://github.com/OfficeDev/office-ui-fabric-react/wiki/Accessibility-Troubleshooting.

**Comment by arunarokk** [comment]
I have created a narrator bug , both narrator & edge team posted their response as below, Here is the reference : https://microsoft.visualstudio.com/OS/OS%20Team/_workitems/edit/18493419

Narrator team:
Actually, the state is announced as part of advanced information.

That said, Narrator does not initially stop on the List control at all. Or it does, but then it tries to set focus to it and all of the sudden it gets a FocusChanged event to the Option element that pulls Narrator inside the List control (again, to the Option element) and interrupts the List announcement to announce the Option control (gee, so much repetition!).

If, from that point, you navigate backwards using item navigation, you will land on the List element and the state will be read.

This is either an issue with the web platform deciding to shift system focus or the framework (React, I assume?). Passing to the WebPlat team as I am not aware of any tools that could help us figure this out without knowing what is happening in Edge. (Although, it would be cool to know if there were any!)

Edge team: 
As soon as you focus the selection box, aria-activedescendant is set to the first option, which means that this element gets the accessibility focus, and that element is not a list. While Narrator could potentially deal with this better, I think everything works as expected in this scenario, and the page just doesn't provide a very good experience.

https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/listbox/listbox-collapsible.html  is the proper ARIA implementation.

**Comment by natalieethell** [comment]
Given what Narrator and Edge said above, I think that this issue might be related to another one being discussed right now: https://github.com/OfficeDev/office-ui-fabric-react/issues/5473. To summarize, it mentions that a dropdown with a placeholder will automatically focus on the first option when opened -- and the selected option if there is already one selected. 

@cliffkoh it looks like you were pretty involved in that discussion. Do you know if this is a behavior we're going to keep? If so, I don't really see another way for this issue to be solved.

**Comment by cliffkoh** [comment]
I have followed up offline with @arunarokk about this. We can have this bug issue be assigned to me and removed off Fabric Shield. It will require deeper investigation from my end but it is very likely the experience is as good as it will be from our end.

**Comment by cliffkoh** [comment]
Per @arunarokk, the current experience is good enough to identify that the Dropdown has options you can select. 

Improvements that can possibly be purused are: 

Required is not announced, required is honored only on the input types
Aria-label is not announced 
Role is not announced.

**Comment by mikewheaton** [comment]
Assigning to @cliffkoh and removing from Fabric Shield as requested.

**Comment by cliffkoh** [comment]
I took another holistic look at this issue today and considered input from @jspurlin as well.

Part of the problem of Fabric's Dropdown is that it neither behaves exactly like an aria `listbox`, nor does it exactly behave like an aria `combobox`. In an attempt to provide a reasonable screen reading experience (for both Jaws and Narrator) that represents all the capabiliites of the Dropdown, in particular the capability to change the selected option through up and down arrow without resulting a popup, it results in the current situation where Narrator + Edge is unable to read the role.

@jspurlin POV is that as with the HTML `<select>` element, the Dropdown should really be of aria role `combobox`. But Fabric's Dropdown lacks the essential combobox requirement described by #6565, where keying in printable characters narrows down and changes the selected option accordingly. So the aria role of `combobox` is inappropriate at the moment because of an incomplete feature set, and in fact would make the Narrator experience even worse.

Perhaps there exists an elegant way to resolve this as-is. Perhaps this will just naturally be supported in Narrator+Edge organically overtime as they ship improvements. But my POV at the moment is that the Narrator experience is just helpful and functional enough, and any further improvements on this front would require implementing #6565 and exploring converting the role to `combobox`. More notably, perhaps the right thing is to combine the Fabric Dropdown and ComboBox components into 1 single component. 

Either way, while noting once again that the experience is good enough, this requires feature-level work to resolve.

(If there exists an apparent elegant solution to this without regressing other Dropdown narration experiences, PRs from community highly welcomed).

**Comment by mikewheaton** [comment]
Thanks for the detailed investigation, @cliffkoh and @jspurlin. I'll change this from a bug to a feature, as it sounds like there is a larger chunk of work to be done here and there are no critical issues at this time.

**Comment by dzearing** [comment]
@jamesedjac @cliffkoh Reading through this, it is unclear that we should be changing anything right now. Can we close this, or is there a recommendation on pursuing more work?

**Comment by cliffkoh** [comment]
@dzearing  This falls under work that will take place with a Combobox/Dropdown rewrite. The work specifically needed for Dropdown in isolation is to implement the functionality Combobox presently has where you can press letters on the keyboard to incrementally narrow down and jump to specific Dropdown action, which will be a big part of unlocking the change of Dropdown to "role=combobox".

**Comment by joschect** [comment]
Since this falls under the refactor found #6030

**Comment by cliffkoh** [comment]
Closing this issue as this is intractable until Dropdown refactor where role moves to "combobox" (in which case this issue is irrelevant). Also FWIW it was determined that this issue is does not block users from having an accessible experience.
